packages/devtools-reps/src/reps/string.js

```diff
@@ -38,41 +39,124 @@ StringRep.propTypes = {
 function StringRep(props) {
   let {
     className,
-    cropLimit,
-    object: text,
-    member,
     style,
+    cropLimit,
+    object,
     useQuotes = true,
     escapeWhitespace = true,
+    member,
     openLink,
     omitLinkHref = true,
   } = props;
 
+  let text = object;
+
+  const isLong = isLongString(object);
+  const shouldCrop = (!member || !member.open) && cropLimit && text.length > cropLimit;
+
+  if (isLong) {
+    text = maybeCropLongString({
+      shouldCrop,
+      cropLimit
+    }, text);
+  }
+
+  text = formatText({
+    useQuotes,
+    escapeWhitespace
+  }, text);
+
+  const config = getElementConfig({
+    className,
+    style,
+    actor: object.actor
+  });
+
+  if (!containsURL(text) || isLong) {
```

Comment:
You can reverse the order to get short-circuit evaluation: `isLong || !containsURL(text)`. Also consider

```js
if (!isLong) {
  if (containsURL(text)) {
    return span(config, ...getLinkifiedElements(/* ... */));
  }
  text = maybeCropString(/* ... */);
}
return span(config, text);
```

Reply:
Good point. 👆